packages/back-end/src/controllers/datasources.ts

```diff
@@ -288,6 +288,12 @@ export async function deleteDataSource(
     );
   }
 
+  if (org.settings?.defaultDataSource === datasource.id) {
+    throw new Error(
+      "Error: This is the default data source for your organization. You must select a new default data source in your Organization Settings before deleting this one."
+    );
+  }
```

Comment:
It seems like we can do this check much earlier and save ourselves a few DB calls

Reply:
Nice catch. Thank you!